python/ml4ir/base/model/scoring/interaction_model.py

```diff
@@ -51,21 +49,19 @@ def __init__(
         self.feature_layer_map.add_fns(feature_layer_keys_to_fns)
 
     def feature_layer_op(self, inputs: Dict[str, Input]):
-        if self.tfrecord_type == TFRecordTypeKey.EXAMPLE:
-            train_features, metadata_features = define_example_feature_layer(
-                feature_config=self.feature_config, feature_layer_map=self.feature_layer_map
-            )(inputs)
-        else:
-            train_features, metadata_features = define_sequence_example_feature_layer(
-                feature_config=self.feature_config,
-                feature_layer_map=self.feature_layer_map,
-                max_sequence_size=self.max_sequence_size,
-            )(inputs)
+        train_features, metadata_features = define_feature_layer(
+            feature_config=self.feature_config,
+            tfrecord_type=self.tfrecord_type,
+            feature_layer_map=self.feature_layer_map,
+        )(inputs)
 
         return train_features, metadata_features
 
     def transform_features_op(self, train_features, metadata_features):
-        # TODO: Make train_features a dictionary
+        # Sorting so that we control the order
```

Comment:
This allows users to do more complex customization if needed.

Reply:
We need to add a comment as such, and why we have created transform_features_op.